Common/Securities/SecurityManager.cs

```diff
@@ -315,25 +315,15 @@ protected virtual void OnCollectionChanged(NotifyCollectionChangedEventArgs chan
             bool addToSymbolCache = true,
             bool isFilteredSubscription = true)
         {
-            if (!subscriptionDataTypes.Any())
-            {
-                throw new ArgumentNullException(nameof(subscriptionDataTypes), "At least one type needed to create security");
-            }
-
             // add the symbol to our cache
             if (addToSymbolCache) SymbolCache.Set(symbol.Value, symbol);
 
             // Add the symbol to Data Manager -- generate unified data streams for algorithm events
+            var configs = subscriptionManager.SubscriptionDataConfigService.Add(symbol, resolution, fillDataForward,
```

Comment:
awesome!

Reply:
we'll have to come back later and see what method args we can clean up -- yay! cleaning dependencies!

Reply:
Totally agree, I've also opened a new card for this https://github.com/QuantConnect/Lean/projects/5#card-13409112 . I'm thinking we could even add an overload which does not create the `SubscriptionDataConfigs` at all.